cscs-checks/microbenchmarks/gpu/gpu_burn/src/gpu_burn.cu

```diff
@@ -39,553 +42,369 @@
-            static const double betaD = 0.0;
-
-            for (size_t i = 0; i < d_iters; ++i) {
-                if (d_doubles)
-                    checkError(cublasDgemm(d_cublas, CUBLAS_OP_N, CUBLAS_OP_N,
-                                           SIZE, SIZE, SIZE, &alphaD,
-                                           (const double*)d_Adata, SIZE,
-                                           (const double*)d_Bdata, SIZE,
-                                           &betaD,
-                                           (double*)d_Cdata + i*SIZE*SIZE, SIZE), "DGEMM");
-                else
-                    checkError(cublasSgemm(d_cublas, CUBLAS_OP_N, CUBLAS_OP_N,
-                                           SIZE, SIZE, SIZE, &alpha,
-                                           (const float*)d_Adata, SIZE,
-                                           (const float*)d_Bdata, SIZE,
-                                           &beta,
-                                           (float*)d_Cdata + i*SIZE*SIZE, SIZE), "SGEMM");
-            }
-        }
-
-        void compare() {
-            int faultyElems;
-            checkError(cuMemsetD32(d_faultyElemData, 0, 1), "memset");
-            dim3 block(g_blockSize,g_blockSize);
-            dim3 grid(SIZE/g_blockSize,SIZE/g_blockSize);
-            //checkError(cuLaunchGrid(d_function, SIZE/g_blockSize, SIZE/g_blockSize), "Launch grid");
-            if(d_doubles)
-                compareD<<<grid,block>>>((double*)d_Cdata,(int*)d_faultyElemData,(size_t)d_iters);
-            else
-                compareF<<<grid,block>>>((float*)d_Cdata,(int*)d_faultyElemData,(size_t)d_iters);
-
-            checkError(cuMemcpyDtoH(&faultyElems, d_faultyElemData, sizeof(int)), "Read faultyelemdata");
-            if (faultyElems) {
-                d_error += (long long int)faultyElems;
-                printf("WE FOUND %d FAULTY ELEMENTS from GPU %d\n", faultyElems, d_devNumber);
-            }
-        }
-
-        private:
-            bool d_doubles;
-            int d_devNumber;
-            size_t d_iters;
-            size_t d_resultSize;
-
-            long long int d_error;
-
-            static const int g_blockSize = 16;
-
-            CUdevice d_dev;
-            CUcontext d_ctx;
-            CUmodule d_module;
-            CUfunction d_function;
-
-            CUdeviceptr d_Cdata;
-            CUdeviceptr d_Adata;
-            CUdeviceptr d_Bdata;
-            CUdeviceptr d_faultyElemData;
-
-            cublasHandle_t d_cublas;
-};
+        // Failure checking
+        if (err)
+            return -1;
 
-// Returns the number of devices
-int initCuda() {
-    checkError(cuInit(0));
-    int deviceCount = 0;
-    checkError(cuDeviceGetCount(&deviceCount));
+        // Get the time difference and return the flops
+        std::chrono::duration<double> diff = end-start;
+        double Gflops = (double)(iters*reps) * ((double)(OPS_PER_MUL)/diff.count()/1000.0/1000.0/1000.0);
```

Comment:
The following, I think, should be enough. The operations will be promoted to double due to `diff.count()`.

```suggestion
        double Gflops = 1e-9 * iters * reps * OPS_PER_MUL * diff.count();
```